src/proxy/inbound.rs

```diff
@@ -107,41 +108,61 @@ impl Inbound {
             }
         }
     }
+
+    /// handle_inbound serves an inbound connection with a target address `addr`.
+    pub(super) async fn handle_inbound(
+        request_type: InboundConnect,
+        addr: SocketAddr,
+    ) -> Result<(), std::io::Error> {
+        let stream = tokio::net::TcpStream::connect(addr).await;
+        match stream {
+            Err(err) => {
+                warn!("connect to {} failed: {}", addr, err);
+                Err(err)
+            }
+            Ok(stream) => {
+                let mut stream = stream;
+                tokio::task::spawn(async move {
+                    match request_type {
+                        InboundConnect::DirectPath(mut incoming) => {
+                            tokio::io::copy_bidirectional(&mut incoming, &mut stream)
+                                .await
+                                .expect("internal server copy");
+                        }
+                        Hbone(req) => match hyper::upgrade::on(req).await {
+                            Ok(mut upgraded) => {
+                                super::copy_hbone("hbone server", &mut upgraded, &mut stream)
```

Comment:
How is copy_hbone different from copy_bidirectional

Reply:
It has its own buffer. We could probably clean it up. However, pretty soon we are going to need more control than `copy_bidirectional` gives us is my guess.

Reply:
Things like reporting metrics, etc

Reply:
Also note I didn't add this in the PR, it was always used - just moving code around